component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticatorConstants.java

```diff
@@ -117,6 +117,8 @@ public class EmailOTPAuthenticatorConstants {
     public static final String PASS_SP_NAME_TO_EVENT = "passSPNameToEvent";
 
     public static final String SCREEN_VALUE = "&screenValue=";
+    public static final String ERROR_TOKEN_EXPIRED_EMAIL_SENT = "&authFailure=true&authFailureMsg=token.expired.email.sent";
```

Comment:
Why do we need two states ? What is the current behaviour ?

Reply:
With the default configuration, OTPs will be sent to the email automatically if the user submits an OTP after the OTP has expired.
But there is a config `disableOTPResendOnFailure = true`, which will disable the automated email sending in case a user submits an expired otp. In this case, the user can only obtain the OTP by clicking the `Resend Code` button.
That's why we have these two states which will display error messages for expired OTP by considering whether the OTP has been sent again or not automatically.
1) `The code entered is expired. Please check inbox for the new OTP.`
2) `The code entered is expired. Click Resend Code to continue.`

Reply:
1) so did we have two different error messages in those two cases before ? 
2) In other OTP flow, do we say it's expired ? what if we give an invalid one ?

Reply:
1) `so did we have two different error messages in those two cases before ?` Actually, we haven't even introduced a specific error in expired otp scenarios before. (Thats what, this patch is all about to introduce a specific error incase of expired OTPs)
2) `In other OTP flow, do we say it's expired? what if we give an invalid one ?` In SMS OTP flow, we say it's expired in case the user submits an expired OTP. And, if we are giving an invalid one, it will just display a generic error message of Authentication Failure.

Reply:
So with new improvements, email OTP will also will behave the same ? If expired, it will say expired and say whther a new one is sent or not.. if it's invalid, it will give a generic error message ?

Reply:
Yes, but we don't have a config in SMS OTP that sends automated OTPs when submitting expired tokens. Rather, it will only send emails upon clicking the `Resend code` button. Therefore, this division of error messages will be not there in SMS OTP. 
Other than that, if the submitted OTP is invalid, it will give a generic error msg in sms and email OTPs.

Reply:
AFAIK, Now the EmailOTP and SMSOTP is not behave in a consistent way.
- In SMSOTP if the OTP is expired, the following error message will be displayed and SMSOTP doesn't have the functionality to sent an OTP automatically if the OTP is expired/invalid. 
`The code entered is expired. Click Resend Code to continue.`
- But EmailOTP has the functionality to send the OTP automatically when the given OTP is expired or invalid and in the UI, it shows the `Resend code ` option as well.  With this `disableOTPResendOnFailure = true` config, we can turn off the the behaviour to sent an OTP automatically.
- In the case of expired/invalid scenario, Since now EmailOTP has two options to sent an OTP (Automatically / via Resend Code option) , better to use different error message.
- However, here the confusion is, if the OTP being sent automatically if it is expired, why should we need to have the `Resend code ` option in the UI? IMO, this `Resend code ` needs to be shown in the UI if the OTP is not configured to sent automatically. This is a separate issue, we can fix this in master if its not added purposefully.

Reply:
@Kanapriya yes, we can add that as an improvement to the master. But for the update we can go ahead with this.